doc/devel/coding_guide.rst

```diff
@@ -7,12 +7,12 @@ Coding guidelines
 We appreciate these guidelines being followed because it improves the readability,
 consistency, and maintainability of the code base.
 
-API changes
-===========
 
-If you are adding new features, changing behavior or function signatures, or
-removing classes, functions, methods, or properties, please see the :ref:`api_changes`
-guide.
+.. admonition:: API changes
+
+  If you are adding new features, changing behavior or function signatures, or removing
```

Comment:
Semi-OT: new features are not *API changes*. The content of that section is not relevant for new features. I propose to remove "new features" here.

On a side note, new features means new API, which should be in line with our API policy. Unfortunately, we haven't written down anything for that, so can't link anywhere. Informally: Use general best practices and be consistent with our current API. - Maybe that's something to write out more explicitly later. OTOH, I'm unclear how helpful an explicit document on that would be for contributors. We have to review new API anyway. Would PRs become much better API-wise if we had such a document? I doubt it. So IMHO for now, API design guidelines are not worth the effort - both for us writing and for contributors reading.

Reply:
Oh! :) In that case I suggest to change

- Page title: "API changes" -> "API guidelines" (in analogy to "Coding guidelines")
- "Announce changes, deprecations, and new features" -> "Announce new features, changes and deprecations", i.e. changed order:  New features are more common (at least for contributions of non-core devs) and thus should be mentioned first.
- Move the section "Adding new features" before "Deprecation" - with the same argument as above.

But that should be a separate PR.